pubsub/gcloud/aio/pubsub/subscriber_message.py

```diff
@@ -1,75 +1,63 @@
+import base64
 import datetime
+import json
 from typing import Any
+from typing import Awaitable
+from typing import Callable
 from typing import Dict
-from typing import List
-from typing import Optional
 
-from google.cloud.pubsub_v1.subscriber.message import Message
+
+def parse_publish_time(publish_time: str) -> datetime.datetime:
+    try:
+        return datetime.datetime.strptime(
+            publish_time, '%Y-%m-%dT%H:%M:%S.%f%z')
+    except ValueError:
+        return datetime.datetime.strptime(
+            publish_time, '%Y-%m-%dT%H:%M:%S%z')
 
 
 class SubscriberMessage:
-    def __init__(self, *args: List[Any],
-                 google_cloud_message: Message = None,
-                 **kwargs: Dict[str, Any]) -> None:
-        if google_cloud_message:
-            self._message = google_cloud_message
-            return
-        self._message = Message(*args, **kwargs)
+    def __init__(self, ack_id: str, message_id: str,
+                 publish_time: 'datetime.datetime',
+                 data: Dict[str, Any],
+                 attributes: Dict[str, Any]):
+        self.ack_id = ack_id
+        self.message_id = message_id
+        self.publish_time = publish_time
+        self.data = data
+        self.attributes = attributes
+        self._callback = self._default_callback()
 
     @staticmethod
-    def from_google_cloud(message: Message) -> 'SubscriberMessage':
-        return SubscriberMessage(google_cloud_message=message)
-
-    @property
-    def google_cloud_message(self) -> Message:
-        return self._message
-
-    @property
-    def message_id(self) -> str:  # indirects to a Google protobuff field
-        return str(self._message.message_id)
-
-    def __repr__(self) -> str:
-        return repr(self._message)
-
-    @property
-    def attributes(self) -> Any:  # Google .ScalarMapContainer
-        return self._message.attributes
-
-    @property
-    def data(self) -> bytes:
-        return bytes(self._message.data)
-
-    @property
-    def publish_time(self) -> datetime.datetime:
-        published: datetime.datetime = self._message.publish_time
-        return published
-
-    @property
-    def ordering_key(self) -> str:
-        return str(self._message.ordering_key)
+    def from_api_dict(received_message: Dict[str, Any]
```

Comment:
Please this method `from_repr()` to match the other `gcloud-aio` APIs. See [an example](https://github.com/talkiq/gcloud-aio/blob/67948064935b592a1e26c3c14039c98be93b7d0a/datastore/gcloud/aio/datastore/key.py#L27-L41). Similarly, building a `to_repr()` for convenience while you're here would be fantastic.

Reply:
omg, I've just realized that I'm doing `json.loads` even though pubsub message can contain anything.
Also, will do repr things

Reply:
Yeah, that one is a Vi-specific thing; we just-so-happen to only send around json-formatted messages.